### PR TITLE
v3.0.0-beta2

### PR DESCRIPTION
**Pull request by Sojaner**
The new push request with the new file names and required changes.

I have updated the Angular package reference also, just remember to upload the new `npm-lock.json` file for the Angular project after you published the `3.0.0-beta2` package on `npm`. :)

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -10,11 +10,11 @@
       "integrity": "sha512-ciiioYMLdo16ShmfHBXJBOFm3xPC4AuwO4xeRpFeHz7WK9PYsWCmigagG2XyzZpubK4a3qNKoUBDhbzHfa50LQ==",
       "dev": true,
       "requires": {
-        "acorn": "5.7.2",
-        "css": "2.2.3",
-        "normalize-path": "2.1.1",
-        "source-map": "0.6.1",
-        "through2": "2.0.3"
+        "acorn": "^5.0.3",
```
Comment (ThomasArdal) [inline package-lock.json:13]: Do you know why there are a lot of version updates her?

**Comment by ThomasArdal** [comment]
I think I'm a little confused now. On your branch, you have the elmahio.js files in both the root and in the src dir. Would like to keep the one in src. There are no changes in the one in root, right?

**Comment by Sojaner** [comment]
Well, actually I reverted the `.d.ts` to its original one, therefore I have
not changed any files at all.

I have only copied the `.js` file and its `.map` file to the root and kept
the `.d.ts` file as it was.

I would keep the `src` folder as is, because that’s obviously the source,
and people will refer to the folder in case they need the source code.

Now we have to decide if we don’t want the `dist` folder and would want to
move everything from that folder to the root or if we want to serve the
`.js` files from that folder and create a `types` folder and serve the
`.d.ts` from that folder, or mix serving the `.js` files from the `dist`
folder and the `.d.ts` file from the root.

You make the decision and let me know. :)

**Comment by ThomasArdal** [comment]
When looking at [this](https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html) and a couple of other repos, I think that we should have the following structure:

- dist
  - elmahio.js (copied here by Gulp)
  - elmahio.js.map (generated and copied here by Gulp)
  - elmahio.min.js (generated and copied here by Gulp)
  - elmahio.min.js.map (generated and copied here by Gulp)
- src
  - elmahio.js
- typescript
  - elmahio.d.ts

The `package.json` would then include something like this:

```json
{
  ...
  "version": "3.0.0-beta2",
  "main": "./dist/elmahio.js",
  "types": "./typescript/elmahio.d.ts",
  ...
}
```

I know it's pretty much like today, but with the TypeScript mappings in a sub-dir. I think the important part of the PR, is to show how to include and use this through Angular. It would be awesome with a sample using WebPack. Gotta look into that at some point.

File: package-lock.json (hunk 1 above)
Comment (Sojaner) [reply package-lock.json:13]: When restoring the dependencies using `npm` it shows a lot of vulnerability issues. Maybe it tries to fix some of them by updating them.

Actually, the Angular project is like stone-age old. ;)

I think it's better to replace the ASP.NET-Angular sample with a pure, updated Angular project altogether. :)

**Comment by Sojaner** [comment]
> When looking at [this](https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html) and a couple of other repos, I think that we should have the following structure:
> 
> * dist
>   
>   * elmahio.js (copied here by Gulp)
>   * elmahio.js.map (generated and copied here by Gulp)
>   * elmahio.min.js (generated and copied here by Gulp)
>   * elmahio.min.js.map (generated and copied here by Gulp)
> * src
>   
>   * elmahio.js
> * typescript
>   
>   * elmahio.d.ts
> 
> The `package.json` would then include something like this:
> 
> ```json
> {
>   ...
>   "version": "3.0.0-beta2",
>   "main": "./dist/elmahio.js",
>   "types": "./typescript/elmahio.d.ts",
>   ...
> }
> ```
> 
> I know it's pretty much like today, but with the TypeScript mappings in a sub-dir. I think the important part of the PR, is to show how to include and use this through Angular. It would be awesome with a sample using WebPack. Gotta look into that at some point.

I will update the repo when I have time. :)

File: package-lock.json (hunk 1 above)
Comment (ThomasArdal) [reply package-lock.json:13]: Totally agree. I created the Angular project using the template available in Visual Studio 2017. I think it's fine to have a sample looking like the template. But a full sample using updated packages, webpack, etc., would be awesome.

**Comment by ThomasArdal** [comment]
I kind of lost track, but we are back to only changing the Angular sample, right? I believe the naming and file structure of my proposal, matches that of today. Besides the typescript folder, but that's the only change as I see it.

**Comment by Sojaner** [comment]
Yes! You can just go ahead and update the `package.json` and publish the beta2 version.

I can create the Angular project sample afterwards if you want me to. :)

**Comment by ThomasArdal** [comment]
I would like the changes you've made in the current Angular sample too. But I can make these manually if it's ok with you?

An Elmah.Io.Js.AngularWebpack sample in a new PR would be really awesome when you have time!

**Comment by Sojaner** [comment]
Absolut! ;) :D Varsågod. :)

I will create the Angular project as soon as I can. :)

**Comment by ThomasArdal** [comment]
Tack så mycket ;)

**Comment by ThomasArdal** [comment]
Really hope that I managed to move all of the changes we've agreed on. In master now :smile:

**Pull request closed without merging**